### PR TITLE
Add product list device navigation

### LINKED ISSUE #8: Add product list subnavigation
## ✨ Updated

This idea has evolved into a more structured sub-navigation for product lists:

Screenshot 2022-01-12 at 09 05 59

So the task is now to implement this new navigation. 

- [x] Add "Guides" link to device page (e.g. https://www.ifixit.com/Device/iPhone_X)
- [x] Add "Answers" link to answers page (e.g. https://www.ifixit.com/Answers/Device/iPhone_X)

This sub-navigation should be recreated across device and answers page too, but that work should be tracked separately.

> ## Outdated description
> The collection page should show a section titled "Model history" like this:
> 
> Screen Shot 2021-05-10 at 18 26 43
> 
> ## ❓Open questions
> 
> - Is this applicable to every collection?
> - What API should we use to get the content of the section?
> 
> ## UPDATE
> 
> - [x] Add text field `deviceTitle` in Strapi collection
> - [x] Decide where to put the link
> - [x] Create the link following this pattern: `https://www.ifixit.com/Device/<DeviceTitle>` where `<DeviceTitle>` should be replaced with the device title with spaces replaced by `_` and then it should be url encoded.

### PR DESCRIPTION
**Pull request by dhmacs**
Closes #8 

## Changelog

- Add a sub navigation to the product list page
- Move breadcrumbs into the subnavigation
- Add device navigation (parts, guides, answers)

## QA

1. Visit the Vercel PR preview
2. Visit parts product list page (`/store/parts`)
3. Verify that "guides" and "answers" links lead to guides home and answers home respectively
4. Visit another product list page (e.g. `Vacuum Parts`) and verify that "guides" and "answers" links points to the device guides and answers pages

### CODE REVIEW

**Comment by deltuh-vee** [comment]
QA 🎬 
* "guides" and "answers" links lead to guides home and answers home, respectively
* device specific parts pages have "guides" and "answers" links leading to device specific guide and answers pages, respectively

File: frontend/components/product-list/ProductListDeviceNavigation.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   if (isRootProductList) {
+      guideUrl = `${IFIXIT_ORIGIN}/Guide`;
+      answersUrl = `${IFIXIT_ORIGIN}/Answers`;
+   } else if (productList.deviceTitle && productList.deviceTitle.length > 0) {
+      guideUrl = `${IFIXIT_ORIGIN}/Device/${snakeCase(
+         productList.deviceTitle
+      )}`;
+      answersUrl = `${IFIXIT_ORIGIN}/Answers/Device/${snakeCase(
+         productList.deviceTitle
+      )}`;
```
Comment (sterlinghirsh) [inline frontend/components/product-list/ProductListDeviceNavigation.tsx:27]: I don't think we need to change the case here, but we will want to replace space with underscore.

Comment (dhmacs) [reply frontend/components/product-list/ProductListDeviceNavigation.tsx]: @sterlinghirsh fixed 👍

**Comment by deltuh-vee** [comment]
Carry over QA 🎬

**Review by masonmcelvain — APPROVED** [review]
CR :v

**Comment by cdcline** [comment]
CR 👍

**Pull request merged**